smithy-model/src/main/java/software/amazon/smithy/model/traits/EnumDefinition.java

```diff
@@ -108,6 +114,93 @@ public static EnumDefinition fromNode(Node node) {
         return builder.build();
     }
 
+    /**
+     * Converts an enum definition to the equivalent enum member shape.
+     *
+     * @param parentId The {@link ShapeId} of the enum shape.
+     * @param synthesizeName Whether to synthesize a name if possible.
+     * @return An optional member shape representing the enum definition,
+     *         or empty if conversion is impossible.
+     */
+    public Optional<MemberShape> asMember(ShapeId parentId, boolean synthesizeName) {
+        String name;
+        if (!getName().isPresent()) {
+            if (canConvertToMember(synthesizeName)) {
+                name = getValue().replaceAll("[-.:/\\s]+", "_");
+            } else {
+                return Optional.empty();
+            }
+        } else {
+            name = getName().get();
+        }
+
+        try {
+            MemberShape.Builder builder = MemberShape.builder()
+                    .id(parentId.withMember(name))
+                    .target(UnitTypeTrait.UNIT)
+                    .addTrait(EnumValueTrait.builder().stringValue(value).build());
+
+            getDocumentation().ifPresent(docs -> builder.addTrait(new DocumentationTrait(docs)));
+            if (!tags.isEmpty()) {
+                builder.addTrait(TagsTrait.builder().values(tags).build());
+            }
+            if (deprecated) {
+                builder.addTrait(DeprecatedTrait.builder().build());
+            }
+
+            return Optional.of(builder.build());
+        } catch (ShapeIdSyntaxException e) {
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Converts an enum definition to the equivalent enum member shape.
+     *
+     * This is only possible if the enum definition has a name.
+     *
+     * @param parentId The {@link ShapeId} of the enum shape.
+     * @return An optional member shape representing the enum definition,
+     *         or empty if conversion is impossible.
+     */
+    public Optional<MemberShape> asMember(ShapeId parentId) {
+        return asMember(parentId, false);
+    }
+
+    /**
+     * Determines whether the definition can be converted to a member.
+     *
+     * @param withSynthesizedNames Whether to account for name synthesization.
+     * @return Returns true if the definition can be converted.
+     */
+    public boolean canConvertToMember(boolean withSynthesizedNames) {
+        return getName().isPresent() || (withSynthesizedNames && CONVERTABLE_VALUE.matcher(getValue()).find());
+    }
+
+    /**
+     * Converts an enum member into an equivalent enum definition object.
+     *
+     * @param member The enum member to convert.
+     * @return An {@link EnumDefinition} representing the given member.
+     */
+    public static EnumDefinition fromMember(MemberShape member) {
```

Comment:
These newly introduced methods are only ever used in the model transform right? I kinda wish these were hidden away in that transform and not publicly callable. Is that possible, or are these used elsewhere?

Reply:
It's used in the actual shape, which has a `fromStringShape` static method, but we could also just move that.

Reply:
The more hidden this is the better IMO as that allows us to change it more over time.

Reply:
I did end up still needing to expose a method to ask whether a shape can be converted, but everything else was able to be hidden away on the enum shape itself.